src/include/distributed/multi_client_executor.h

```diff
@@ -72,6 +75,28 @@ typedef enum
 } BatchQueryStatus;
 
 
+/* Enumeration to track what the task associated with a connection is waiting for */
+typedef enum WaitStatus
+{
+	WAIT_BUSY, /* no IO required, ready to be processed further */
```

Comment:
And here we'd have `TASK_CONN_READY`, `TASK_CONN_ERRORED`, `TASK_CONN_WAIT_READ`, and `TASK_CONN_WAIT_WRITE`